### PR TITLE
binaryxml: fix stringpool insertion padding

### PR DESCRIPTION
**Pull request by mhils**
I previously had some APKs that failed after patching, this fixes it. 😃

### CODE REVIEW

File: frida_tools/apk.py [hunk 1]
```diff
@@ -299,10 +299,11 @@ def append_str(self, add: str) -> int:
...
 
         if self.utf8:
-            # UTF-16 length (ignored)
-            chunk_data.extend(struct.pack("<B", len(add)))
-            # UTF-8 length
+            assert len(add.encode("utf-8")) < 128  # multi-byte len strings not supported yet
+            # the length of the string in characters
             chunk_data.extend(struct.pack("<B", len(add)))
+            # the length of the UTF-8 encoding of the string in bytes
+            chunk_data.extend(struct.pack("<B", len(add.encode("utf-8"))))
```
Comment (oleavr) [inline frida_tools/apk.py:306]:
```suggestion
            length_in_characters = len(add)
            length_in_bytes = len(add.encode("utf-8"))
            chunk_data.extend(struct.pack("<BB", length_in_characters, length_in_bytes))
```

(Should avoid comments if the code can be expressed such that they're not needed.)

File: frida_tools/apk.py [hunk 2]
```diff
@@ -313,6 +314,9 @@ def append_str(self, add: str) -> int:
             # Insert a UTF-16 NUL
             chunk_data.extend([0, 0])
 
+        # pad to a multiple of 4 bytes
+        chunk_data.extend([0] * (4 - len(chunk_data) % 4))
```
Comment (oleavr) [inline frida_tools/apk.py:318]:
```suggestion
        alignment_padding = [0] * (4 - len(chunk_data) % 4)
        chunk_data.extend(alignment_padding)
```

Comment (oleavr) [reply frida_tools/apk.py:318]: Won't this add 4 bytes of padding when `chunk_data` is already a multiple of four?

**Review by oleavr — CHANGES_REQUESTED** [review]
Ahh nice, thanks! 🙌 (And apologies for the absurd delay, finally now starting to catch up on things after life got super-busy.)

File: frida_tools/apk.py (hunk 2 above)
Comment (mhils) [reply frida_tools/apk.py]: It did, fixed! 😃

**Comment by oleavr** [comment]
Thanks! 🙌

**Pull request merged**